### ISSUE TITLE
equals method overriding is missing 

### ISSUE THREAD

**Issue by ayoub-benali** [tresata/spark-kafka#3]
``` scala
private class KafkaPartition(rddId: Int, override val index: Int, val partition: Int, val startOffset: Long, val stopOffset: Long, val leader: Option[Broker])
    extends Partition {
  override def hashCode: Int = 41 * (41 + rddId) + index
}
```

Source code [here](https://github.com/tresata/spark-kafka/blob/master/src/main/scala/com/tresata/spark/kafka/KafkaRDD.scala#L29)

objects, KafkaPartitions in this case, which are .equals() must have the same .hashCode()

The problem with "equals" method implementation missing is that now two Objects with the same hashcode will be considered different. 
- Is that done in purpose ?
- why is hashCode overridden this way using rddId and index ?

If that is a mistake I can issue a PR with equals being implemented comparing rddId and index which are used in hashCode

**Comment by koertkuipers** [comment]
sorry i missed this.

i dont remember why hashCode is defined this way. i think i took it from another partition implementation. you are right that it would be good practice to override equals when overriding hashCode

**Comment by koertkuipers** [comment]
fixed in e74ec005435d23227a1bc6cb42e2e98141f70458
